fedimint-server/src/consensus/mod.rs

```diff
@@ -547,6 +549,47 @@ impl FedimintConsensus {
         Ok(())
     }
 
+    pub async fn await_transaction_status(
+        &self,
+        txid: TransactionId,
+    ) -> Option<crate::outcome::TransactionStatus> {
+        let accepted_key = AcceptedTransactionKey(txid);
+        let rejected_key = RejectedTransactionKey(txid);
+        tokio::select! {
+            accepted = self.db.wait_key(&accepted_key) => {
+                let mut outputs = Vec::new();
+                let accepted = accepted.expect("DB error");
+                // timeout
+                let Some(accepted) = accepted else { return None; };
+                let mut dbtx = self.database_transaction().await;
+                for (out_idx, output) in accepted.transaction.outputs.iter().enumerate() {
+                    let outpoint = OutPoint {
+                        txid,
+                        out_idx: out_idx as u64,
+                    };
+                    let outcome = self
+                        .modules
+                        .module(output.module_key())
+                        .output_status(&mut dbtx, outpoint)
+                        .await
+                        .expect("the transaction was processed, so should be known");
+                    outputs.push((&outcome).into())
+                }
+
+                Some(crate::outcome::TransactionStatus::Accepted {
+                    epoch: accepted.epoch,
+                    outputs,
+                })
+
+            }
+            rejected = self.db.wait_key(&rejected_key) => {
```

Comment:
Are the timers setup so that wait_key will always time out after it is guaranteed that a transaction is rejected? (i.e its not possible for wait_key to timeout but the transaction simple hasn't been accepted yet)

Reply:
nope, timeouts are not related to transaction at all.

Reply:
So on a timeout we are returning TransactionStatus::Rejected. When reading the code, that sounds like the transaction was unsuccessful, but actually it is possible that the transaction just hasn't processed yet. Am I interpreting TransactionStatus::Rejected correctly?

Reply:
Nope, we return `None` in case of timeout. (let .. else statement)